api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java

```diff
@@ -32,6 +33,11 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  private final static NoneShardSpec INSTANCE = new NoneShardSpec();
```

Comment:
@nishantmonu51 maybe we can mark the constructor as deprecated then, so we can remove it in the future.